src/main.ts

```diff
@@ -0,0 +1,118 @@
+    });
+}
+
+async function setTrivyEnvVariables() {
+    for (let key in process.env) {
+        trivyEnv[key] = process.env[key] || "";
+    }
+
+    const username = core.getInput("username");
+    const password = core.getInput("password");
+    if (username && password) {
+        trivyEnv["TRIVY_AUTH_URL"] = "https://registry.hub.docker.com";
+        trivyEnv["TRIVY_USERNAME"] = username;
+        trivyEnv["TRIVY_PASSWORD"] = password;
+    }
+
+    trivyEnv["TRIVY_EXIT_CODE"] = "1";
+
+    try {
+        const whitelistFilePath = core.getInput("whitelist-file");
+        const whitelistFileBranch = core.getInput("whitelist-file-branch");
+        if (whitelistFilePath) {
+            const whitelistFileLoc = await getWhitelistFileLoc(whitelistFilePath, whitelistFileBranch);
+            trivyEnv["TRIVY_IGNOREFILE"] = whitelistFileLoc;
+        }
+    } catch (error) {
+        throw new Error(util.format("Could not download whitelist file. Error: %s", error.message));
+    }
+}
+
+async function setDockleEnvVariables() {
+    for (let key in process.env) {
+        dockleEnv[key] = process.env[key] || "";
+    }
+
+    const username = core.getInput("username");
+    const password = core.getInput("password");
+    if (username && password) {
+        dockleEnv["DOCKLE_AUTH_URL"] = "https://registry.hub.docker.com";
+        dockleEnv["DOCKLE_USERNAME"] = username;
+        dockleEnv["DOCKLE_PASSWORD"] = password;
+    }
+
+    dockleEnv["DOCKLE_EXIT_CODE"] = "1";
+}
+
+async function run(): Promise<void> {
+    try {
+        const trivyPath = await getTrivy();
+        console.log(util.format("Trivy executable found at path ", trivyPath));
+
+        await setTrivyEnvVariables();
+
+        const imageName = core.getInput("image-name");
+        const trivyOptions: ExecOptions = {
+            env: trivyEnv,
+            ignoreReturnCode: true
+        };
+        let trivyArgs = [];
+        trivyArgs.push(imageName);
+        const trivyToolRunner = new ToolRunner(trivyPath, trivyArgs, trivyOptions);
+
+        const trivyStatus = await trivyToolRunner.exec();
+
+        const addCISChecks = core.getInput("add-CIS-checks");
+        if (addCISChecks.toLowerCase() == "true") {
+            const docklePath = await getDockle();
+
+            await setDockleEnvVariables();
+
+            const dockleOptions: ExecOptions = {
+                env: dockleEnv,
+                ignoreReturnCode: true
+            };
+
+            let dockleArgs = [];
+            dockleArgs.push(imageName);
+
+            const dockleToolRunner = new ToolRunner(docklePath, dockleArgs, dockleOptions);
+
+            await dockleToolRunner.exec();
+        }
+
+        if (trivyStatus == 0) {
+            console.log("No vulnerabilities were detected in the container image");
+        } else if (trivyStatus == 1) {
+            throw new Error("Vulnerabilities were detected in the container image");
```

Comment:
Should we fail the scan on Dockle failure as well?

Reply:
Checked the spec, we should.
Maybe out of scope of this PR, but there is case of dockle whitelist as well. We will need to close on a format for whitelist.

Reply:
Dockle provides CIS guidelines. They should be considered as best practices to create an image. I don't think we should fail the workflow if those guidelines are not followed. We can take PM input on this though.

Reply:
I agree, we should not fail the action on dockle failure but the output should convey the same though

Reply:
Sounds good. We can start with not failing and discuss with PM once.